cfgov/unprocessed/js/modules/util/validators.js

```diff
@@ -21,8 +21,9 @@ const typeCheckers = require( '../../modules/util/type-checkers' );
  */
 function date( field, currentStatus ) {
   const status = currentStatus || {};
+  // Date regex matches date patterns allowed in cfgov/v1/forms.py FilterableDateField
   const dateRegex =
-    /^\d{2}$|^\d{4}$|^\d{2}\/(?:\d{4}|\d{2})$|^\d{2}\/\d{2}\/\d{4}$/;
+    /^\d{2}$|^\d{4}$|^(?:\d{1}|\d{2})\/(?:\d{4}|\d{2})$|^(?:\d{1}|\d{2})\/(?:\d{1}|\d{2})\/\d{4}$/;
```

Comment:
- The first option `^\d{2}$` matches a solo 2-digit year input string like `16` but not a 4-digit year like `2016`. Should it be `^\d{4}$` instead?
- The second option `^(?:\d{1}|\d{2})\/(?:\d{4}|\d{2})$` looks good and does match `1/16`, `10/16`, `1/2016`, `10/2016`.
- The third option `^(?:\d{1}|\d{2})\/(?:\d{1}|\d{2})\/\d{4}$` matches m/d/y with 4-digit years like `1/10/2016` but doesn't seem to match 2-digit years like `10/1/16` as documented [here](https://github.com/cfpb/cfgov-refresh/pull/4643/files#diff-a20999810035f4df9b089b5ce85e66b6R22) as a valid Python option. Should it be `^(?:\d{1}|\d{2})\/(?:\d{1}|\d{2})\/(?:\d{4}|\d{2})$` instead?

Reply:
- The first option was extraneous. Thanks for calling that out.
- Good call, I forgot to add the two-digit option to the third option. I've added it now.